docs/clientdev/features.rst

```diff
@@ -0,0 +1,156 @@
+itself. For framework intergations this is typically part of middleware or some
+other system.
+
+Setting Attributes
+------------------
+
+Ability to set attributes on events sent to Sentry. They should be configurable
+"globally" on the client itself (so that they are sent with all future events)
+and also per event. Important examples are being able to set the ``release``,
+``environment``, ``tags`` and ``extra``. :ref:`See the attributes page <attributes>`
+for more information and examples.
+
+Automatic Attributes
+--------------------
+
+Automatic addition of useful attributes such as ``tags`` or ``extra``.
+Typically means the SDK hooks into a framework so that it can set attributes
+that are known to be useful for most users.
+
+Context system
+--------------
+
+A system for storing data that has to do with a given context. What context means
+depends on the application, for a web framework it is most likely a single request/response
+cycle. For a mobile application there is often just one single context that represents
+the single user and their actions. Context can be difficult to implement because it often
+has to deal with threads or concurrency and can involve deep integration with frameworks.
+:ref:`See the context page <context>` for more information.
+
+Breadcrumbs
+-----------
+
+Manually record application events (into the current context) during the lifecycle of an
+application. Needs to implement a ring buffer so as not to grow indefinitely. The most recent
+breadcrumbs should be attached to events as they are created.
+
+With deeper framework integration, the automatic recording of some breadcrumbs is possible and
+recommended.
+
+Event Sampling
+--------------
+
+SDKs should allow the user to configure what percentage of events are actually
+sent to the server (the rest should be silently ignored). For example:
+
+.. sourcecode:: python
+
+    sample_rate = options.get('sample_rate', 1.0)
+
+    # assuming random() returns a value between 0.0 (inclusive) and 1.0 (exclusive)
+    if random() < sample_rate:
+        client.send(data)
+
+HTTP 429 Retry-After backoff
+----------------------------
+
+Respect Sentry’s HTTP 429 Retry-After header when the user goes over their limits. Events
+should be dropped during backoff.
+
+Ignore patterns
+---------------
+
+The user should be able to define exceptions that are ignored/dropped based on the
+error message or type.
+
+In-App frames
+-------------
+
+Stack parsing can tell which frames should be identified as part of the user’s application
+(as opposed to part of the language, a library, or a framework), either automatically or by
+user configuration at startup, often declared as a package/module prefix.
+
+Surrounding source in stacktrace
+--------------------------------
+
+Lines of source code to provide context in stacktraces. This is easier in interpreted languages,
+may be hard or impossible in compiled ones.
+
+Stack variable values
+---------------------
```

Comment:
Php for example is actually just the function args which I think is more common to what we can get